ssh-server/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os/exec"
+
+	"github.com/kr/pty"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	serverPort = flag.Uint("port", uint(22), "the port to listen on")
+)
+
+func main() {
+	flag.Parse()
+
+	config, listener, err := startServer()
+	if err != nil {
+		log.Fatalf("Failed to start server: %v", err)
+	}
+
+	for {
+		connection, err := listener.Accept()
+		if err != nil {
+			log.Fatalf("Failed to accept incoming connection: %v", err)
+		}
+
+		go func() {
+			handleConnection(connection, config)
+		}()
+	}
+}
+
+func startServer() (*ssh.ServerConfig, net.Listener, error) {
+	config, err := getConfiguration()
+	if err != nil {
+		return nil, nil, fmt.Errorf("config server: %v", err)
+	}
+
+	serverAddress := fmt.Sprintf(":%d", *serverPort)
+	listener, err := net.Listen("tcp", serverAddress)
+	if err != nil {
+		return nil, nil, fmt.Errorf("listening for connection: %v", err)
+	}
+	log.Printf("Listening on %s...", serverAddress)
+	return config, listener, nil
+}
+
+func getConfiguration() (*ssh.ServerConfig, error) {
```

Comment:
I think we should probably generate a key internally, actually - otherwise all the servers have the same key.  It maybe doesn't matter since we aren't really using the key anyway.

@gkelly thoughts?